libs/nio/src/main/java/org/opensearch/nio/ServerChannelContext.java

```diff
@@ -70,15 +70,10 @@ public ServerChannelContext(NioServerSocketChannel channel, ChannelFactory<?, ?>
     }
 
     public void acceptChannels(Supplier<NioSelector> selectorSupplier) throws IOException {
-        SocketChannel acceptedChannel = null;
-        try {
-            while ((acceptedChannel = accept(rawChannel)) != null) {
-                NioSocketChannel nioChannel = channelFactory.acceptNioChannel(acceptedChannel, selectorSupplier);
-                acceptor.accept(nioChannel);
-            }
-        } finally {
-            if (acceptedChannel != null)
-                acceptedChannel.close();
+        SocketChannel acceptedChannel;
+        while ((acceptedChannel = accept(rawChannel)) != null) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *acceptedChannel*. Execution paths that do not contain closure statements were detected. Ensure *acceptedChannel* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).